sway-fmt/src/traversal_helper.rs

```diff
@@ -163,3 +223,30 @@ fn get_data_field_type(line: &str, iter: &mut Peekable<Enumerate<Chars>>) -> Str
 
     result
 }
+
+#[cfg(test)]
+mod tests {
+    use super::sort_and_filter_use_expression;
+
+    #[test]
+    fn test_sort_and_filter_use_expression() {
+        assert_eq!(sort_and_filter_use_expression("::a::b::c;"), "::a::b::c;");
+        assert_eq!(
+            sort_and_filter_use_expression("::a::c::b::{c, b, ba};"),
+            "::a::c::b::{b,ba,c};"
+        );
+        assert_eq!(
+            sort_and_filter_use_expression("{s,e,l,f,self};"),
+            "{self,e,f,l,s};"
+        );
+        assert_eq!(
+            sort_and_filter_use_expression("a::{d::{f, self}, c, b};"),
+            "a::{b,c,d::{self,f}};"
+        );
+        assert_eq!(
+            sort_and_filter_use_expression("a::b::{c,d::{self,f}};"),
+            "a::b::{c,d::{self,f}};"
+        );
+        assert_eq!(sort_and_filter_use_expression("a::b::{c;"), "a::b::c;");
```

Comment:
`a::b::{c;` is really a syntax error right? I don't think the formatter should change the user code in a non-trivial way like this.

Reply:
Agreed, the parser would barf before we'd see this, but removing the `{}` from `a::b::{c};` is something worth doing.

Reply:
Done. Updated to remove {} when only single element inside.